db/write_batch_internal.h

```diff
@@ -177,12 +177,51 @@ class WriteBatchInternal {
                            uint64_t log_number = 0, DB* db = nullptr,
                            bool concurrent_memtable_writes = false);
 
-  static void Append(WriteBatch* dst, const WriteBatch* src,
-                     const bool WAL_only = false);
+  static Status Append(WriteBatch* dst, const WriteBatch* src,
+                       const bool WAL_only = false);
 
   // Returns the byte size of appending a WriteBatch with ByteSize
   // leftByteSize and a WriteBatch with ByteSize rightByteSize
   static size_t AppendedByteSize(size_t leftByteSize, size_t rightByteSize);
 };
 
+// LocalSavePoint is similar to a scope guard
+class LocalSavePoint {
```

Comment:
I use it in WriteBatchInternal::* as a way to restore WB state if memory is exceeded.

Actually, we can reuse a regular savepoint. I just thought it'd be more efficient/convenient to use this.